src/app/login/Login.tsx

```diff
@@ -4,13 +4,36 @@ import Box from '@mui/material/Box';
 import LockOutlinedIcon from '@mui/icons-material/LockOutlined';
 import Typography from '@mui/material/Typography';
 import Container from '@mui/material/Container';
+import { useEffect } from 'react';
+import { useNavigate } from 'react-router-dom';
 
 import { styles } from './Login.styles';
 import { useLocale } from 'hooks';
 import { LoginForm } from './loginForm/LoginForm';
+import { loginAction } from 'api/actions/auth/authActions';
+import { useMutation } from 'api/hooks';
+import { LoginPayload } from 'api/actions/auth/auth.types';
+import { useAuth } from 'hooks/useAuth/useAuth';
+import { AppRoute } from 'routing/AppRoute.enum';
 
 export const Login = () => {
   const { formatMessage } = useLocale();
+  const { mutate, data, isLoading, isError } = useMutation<LoginPayload>(loginAction);
+  const { setToken, isLoggedIn } = useAuth();
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    if (isLoggedIn) {
+      navigate(AppRoute.home);
+    }
+  }, [isLoggedIn]);
```

Comment:
A może zrobić z tego jakiś ładny komponent? Bo widzisz, na ten moment, jeśli ktoś jest zalogowany, to jednak pokaże mu się strona logowania, a potem go przekieruje.
A można by zrobić komponent `Authenticated` jako wrapper, który w zależności od propsa `shouldAuthenticate: boolean` wyświetlać spinner albo childreny. I miałby w sobie też useEffect z redirectem